codes/dataops/augmennt/augmennt/functional.py

```diff
@@ -168,13 +169,14 @@ def resize(img, size, interpolation='BILINEAR'):
         raise TypeError('img should be numpy image. Got {}'.format(type(img)))
     if not (isinstance(size, int) or (isinstance(size, collections.Iterable) and len(size) == 2)):
         raise TypeError('Got inappropriate size arg: {}'.format(size))
-    
+
     w, h, =  size
     if isinstance(size, int):
         # h, w, c = img.shape #this would defeat the purpose of "size"
-        
+
         if (w <= h and w == size) or (h <= w and h == size):
             return img
+        img = srgb2linear(img)
```

Comment:
As far as I can tell this resize() function can be entered both inside of another converted block. It's probably best to just introduce wrapper types to avoid double conversions anywhere.